\chapter{Other projects and tools}\label{other_projects}

This chapter presents a few other mention-worthy projects and tools that I worked on during my thesis.

\localtableofcontents

\section{Bacterial chromatin and HU}

Initially the main project of my PhD, the study of bacterial chromatin compaction by cryo-EM and cryo-ET is an ongoing project in the MICA group, which builds upon previous fluorescence imaging done by the GenOM team on \textit{D. radiodurans} nucleoids.
The main protein of interest is HU, a nucleoid-associated protein (NAP) present in most bacteria (including \textit{E. coli} and \textit{D. radiodurans}) which is known to be a key player in chromatin compaction and nucleoid morphology throughout the cell cycle.
The structure of dimeric HU has been solved from various bacterial species in either its apo- or DNA-bound forms (\autoref{fig:hu}).
HU is thought to have a histone-like function --- with molecular dynamics simulations supporting the evidence to this binding mode~\cite{hognonMolecularBasesDNA2019} --- but it was also observed forming stacks of dimers that lay parallel to the DNA duplex, suggesting the possibility of multiple modes of interaction between HU dimers and the DNA~\cite{hammelHUMultimerizationShift2016}.
These different modes may explain

\begin{figure}[ht]
    \centering
    \begin{subfigure}[B]{.48\textwidth}
        \centering
        \includegraphics[width=\textwidth]{other/dr_nucleoids.png}
        \caption{\textit{D. radiodurans} heterogeneous and dynamic nucleoid morphology shows variations throughout the cell cycle. Figure taken from \citet{flochCellMorphologyNucleoid2019}.}
        \label{fig:hu_nucleoids}
    \end{subfigure}%
    \hfill
    \begin{subfigure}[B]{.5\textwidth}
        \centering
        \includegraphics[width=\textwidth]{other/hu_struct.png}
        \caption{\textit{Borrelia burgdorferi} HU structure determined by X-ray crystallography. Its DNA-kinking histone-like behavior was replicated by molecular dynamics simulations. Figure taken from \citet{hognonMolecularBasesDNA2019}.}
        \label{fig:hu_structure}
    \end{subfigure}%
    \titledcaption[\textit{D. radiodurans} nucleoids and HU structure]{\textbf{(a)} Nucleoid compaction and morphology in \textit{D. radiodurans} is complex and dynamic, with HU being one of the major drivers of chromatin compaction. \textbf{(b)} HU is thought to have a histone-like behaviour, bending the DNA to form a tight kink.}
    \label{fig:hu}
\end{figure}

To begin investigating DrHU in complex with the DNA, we collected single particle cryo-EM data of an \textit{in vitro} preparation of plasmid DNA and HU, in order to observe the interaction in a free aqueous environment.
The bacterial genomic DNA is large, circular and supercoiled; where previous structural studies used short DNA segments, we opted for a circular dsDNA plasmid as it would be a more appropriate stand-in that would allows us to evaluate how circularity, supercoiling and DNA topology affect HU binding and assembly on DNA.
HU and plasmid DNA were mixed at different ratios, and within a small range of relative concentration (\sim100 HU/plasmid) intriguing spirals (\autoref{fig:hu_spirals}) were formed by the DNA.
Notably, these spirals were never observed in the absence of HU, and were also not observed when using older batches of the plasmid. % TODO: what does this mean? What older batch?

\begin{figure}[ht]
    \centering
    \includegraphics[width=\textwidth]{other/hu_spirals_no_spirals.png}
    \titledcaption[HU-induced DNA spirals]{Spiral formations of DNA are formed in presence of HU only at specific concentration ranges.}
    \label{fig:hu_spirals}
\end{figure}

The very small molecular weight of HU (\sim25 kDa in its dimeric form) meant that picking and processing these particles (even in the expected dimeric or tetrameric forms) would be practically impossible in cryo-EM (see \autoref{fig:et_smallest_particle} for the theoretical limits).
However, due to the relatively high concentration of HU --- \sim100 HU/plasmid corresponds to approximately 1 HU for every 26bp --- the DNA should be densely decorated by HU, unless the protein distribution on the grid is not uniform or big aggregates are forming.
It should therefore be possible to work around the particle size limitation by using the supporting geometry of the DNA filament to do the picking and guide refinement.

This, unfortunately, proved unsuccessful; while we were able to pick and classify on the DNA and several interesting "blobs" next to it (\autoref{fig:hu_classes}), we could never reach resolutions high enough to convincingly identify them as HU.
% TODO: show unsuccessful picking strategies...

\begin{figure}[ht]
    \centering
    \includegraphics[width=\textwidth]{other/hu_classes.png}
    \titledcaption[HU+DNA: representative 2D classes]{Representative classification results from generic particle picks from the filament picker (top) and from picks centered on "blobs" next to the DNA (bottom).}
    \label{fig:hu_classes}
\end{figure}

Classification and refinement are not only made difficult by the small size of HU, but also by the heterogeneous and dynamic nature of the HU-DNA complex.

\subsection{Future perspectives}

To increase our chances to identify and pick HU, a new dataset was collected with a phase plate, which allows to collect data closer to focus while improving the contrast of low spatial fequency information (\fullref{em_ctf}).

Preliminary results on this data --- now being processed by Harald Bernhard, a postdoc in the MICA group --- showed more classes containing what could be HU in a similar conformation to what seen in \autoref{fig:hu_structure} (\autoref{fig:hu_blobs}).

\begin{figure}[ht]
    \centering
    \includegraphics[width=\textwidth]{other/hu_blobs.png}
    \titledcaption[HU+DNA: promising 2D classes]{Promising 2D classes from the phase plate dataset, potentially showcasing DNA in complex with HU.}
    \label{fig:hu_blobs}
\end{figure}

TODO: some hint on irina's ideas?

\section{waretomo}\label{waretomo}

As most cryo-ET users, over time I kept tweaking and improving my processing pipeline.
At present, it consists of three main elements: Warp~\cite{tegunovRealtimeCryoelectronMicroscopy2019} for preprocessing and tomogram and subtomogram reconstruction, AreTomo~\cite{zhengAreTomoIntegratedSoftware2022} for tilt series alignment, and Relion~\cite{scheresRELIONImplementationBayesian2012,zivanovBayesianApproachSingleparticle2022,burtImageProcessingPipeline2024} for STA.

To facilitate batch processing and remove most needs for manual intervention (and potential points of failure), I developed an automated tool which connects Warp to AreTomo: \href{https://gihub.com/brisvag/waretomo}{waretomo}~\cite{gaifasWaretomo2024}.

Specifically, waretomo:
\begin{enumerate}[noitemsep]
    \item automates parsing \texttt{.mdoc} files to provide appropriate inputs to AreTomo
    \item ensures that tilts skipped in Warp are properly handled
    \item adjusts tilt angles in Warp based the otuput of AreTomo to untilt tilted samples (such as lamellae)
    \item optionally reconstructs preliminary tomograms with AreTomo and denoises them with Topaz~\cite{beplerTopazDenoiseGeneralDeep2020}
\end{enumerate}

When possible, each step is performed in parallel (and on GPUs) to maximize resource usage and minimize idle time.
While this tool is available and open source, it's not yet publication-ready and still in alpha, so bugs are expected.

\section{emscan}

In cell-extract cryo-EM~\cite{suBuildRetrieveMethodology2021,kyrilisIntegrativeBiologyNative2019}, a dataset may contain a large variety of proteins and complexes, many of which unknown.
In these non-purified datasets, automating parts of the selection and protein identification processes is essential for practical applications such as structural proteomics of a novel organism.

As a contribution to the PhD project of Eymeline Pageot --- another PhD student in the MICA group under the supervision of Ambroise Desfosses --- I developed \href{https://gihub.com/brisvag/emscan}{emscan}, a napari~\cite{thenaparicommunityNapariMultidimensionalImage2024}-based tool that searches the EMDB database~\cite{thewwpdbconsortiumEMDBElectronMicroscopy2024} for maps whose projections have a high similarity with the input 2D classes.

This allows for example to discard classes from already-solved structures, or to identify 2D classes that potentially belong to different views of a protein whose homologues are present on the EMDB.

% TODO explain how it works

\section{Teamtomo}\label{teamtomo}

Early on in my PhD, Alister Burt and I started the \href{https://teamtomo.org}{teamtomo} project, with the goal of creating a shared, open-source resource for cryo-ET developers, and to encourage the community to collaborate on the development of cryo-ET software within the python ecosystem.

At this time, almost 30 people from several different groups around the world have contributed to libraries and tools in the \href{https://github.com/teamtomo}{teamtomo repositories}, and more are joining our montly meetings where we plan future concerted efforts.

% TODO: expand a bit?

\section{Other napari tools}

During my thesis, napari proved time and time again a powerful library for creating custom interactive visualisation tools.
Some smaller napari-based tools and plugins I contributed to that are worthy of mention are:

\begin{enumerate}
    \item \href{https://github.com/cellcanvas/surforama}{surforama}~\cite{yamauchiSurforamaInteractiveExploration2024}, a tool to explore cryo-ET membrane annotations inspired by membranorama~\cite{tegunovDtegunovMembranorama2024}, fruit of a collaboration with other napari developers and members of the teamtomo community
    \item \href{https://github.com/brisvag/napari-molecule-reader}{napari-molecule-viewer}, a plugin to open PDB and mmCIF files in napari
    \item a \href{https://github.com/napari/napari/blob/main/examples/fourier_transform_playground.py}{fourier transform playground} to interactively explore the relationship between images and their FTs
    \item a \href{https://gist.github.com/brisvag/d6394d05b2f994e083ec279d6976484f}{Euler angle playground} to build an intuition on the various Euler angle conventions and their effects on particle orientations
\end{enumerate}

\section{cs2star}

A small tool, but used widely by our group and others, and later \href{https://sbgrid.org/software/titles/cs2star}{added to SBGrid}~\cite{morinCuttingEdgeCollaboration2013}, \href{https://github.com/brisvag/cs2star}{cs2star}~\cite{gaifasCs2starPy2021} is a small wrapper around the widespread \href{https://github.com/asarnow/pyem}{cryosparc2star.py} which automates a few tedious manual steps normally needed when converting CryoSPARC~\cite{punjaniCryoSPARCAlgorithmsRapid2017} projects to Relion~\cite{scheresRELIONImplementationBayesian2012} ones.

\section{stemia}

\href{Stemia}{https://github.com/brisvag/stemia} is a collection of command-line tools that I developed but that were either too small or too niche to make sense as standalone packages.

\subsection{Out-of-plane angle generation}\label{stemia_angles}
This tool was developed to work on the FtsZ tilted dataset (\fullref{ftsz_tilted}). % TODO: expand

\begin{outline}
\1 teamtomo!
    \2 goal proposition, community efforts
\1 projection matching for eymeline's work
    \2 more native than SPA but more straightforward than STA
    \2 would be cool to go into some technical details here (cross-correlation, projection, etc)
    \2 optimizing for space and performance
    \2 the issues around cross correlation
\1 waretomo
    \2 some suites exist that do the "whole" procedure, but they are monolithic and hard to get in/out for special cases, which occurs frequently in cryoet
    \2 difficulties with metadata wrangling
\end{outline}
